caiman/source_extraction/volpy/spikepursuit.py

```diff
@@ -483,43 +486,38 @@ def denoise_spikes(data, window_length, fr=400,  hp_freq=1, threshold_method='si
     window = np.int64(np.arange(-window_length, window_length + 1, 1))
     locs = locs[np.logical_and(locs > (-window[0]), locs < (len(data) - window[-1]))]
     PTD = data[(locs[:, np.newaxis] + window)]
-    PTA = np.mean(PTD, 0)
+    PTA = np.median(PTD, 0)
+    PTA = PTA - np.min(PTA)
     templates = PTA
 
     # whitened matched filter
     datafilt = whitened_matched_filter(data, locs, window)    
     datafilt = datafilt - np.median(datafilt)
 
     # spikes detected after filter
-    pks2 = datafilt[signal.find_peaks(datafilt, height=None, distance=int(fr/100))[0]]
+    pks2 = datafilt[signal.find_peaks(datafilt, height=None)[0]]
     if threshold_method == 'simple':
         ff1 = -datafilt * (datafilt < 0)
         Ns = np.sum(ff1 > 0)
         std2 = np.sqrt(np.divide(np.sum(ff1**2), Ns)) 
         thresh2 = threshold * std2
-        spikes = signal.find_peaks(datafilt, height=thresh2, distance=int(fr/100))[0]
+        spikes = signal.find_peaks(datafilt, height=thresh2)[0]
+        
         if len(spikes) < min_spikes:
             low_spikes = True
-            if last_round == True:
-                logging.warning(f'last round: less than {min_spikes} spikes are found')
-            else:
-                logging.warning(f'less than {min_spikes} spikes are found, pick top {min_spikes} spikes')
-                thresh2 = np.percentile(pks2, 100 * (1 - min_spikes / len(pks2)))
-                spikes = signal.find_peaks(datafilt, height=thresh2, distance=int(fr/100))[0]
+            logging.warning(f'less than {min_spikes} spikes are found, pick top {min_spikes} spikes')
```

Comment:
I know you didn't write it that way, but can you have it say "were found" rather than "are found"? Also, maybe "picking" rather than "pick".

Reply:
sure